perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -4357,34 +4355,47 @@ private static Map<String, Integer> createMappingOfValues(PerunBean complementar
 
 	/**
 	 * Checks, if principal was authorized by Multi-factor authentication.
-	 * The information is resolved in UserInfoEndpointCall and stored in principal's additionalInformations.
-	 * The timestamp of MFA must not be older than mfa timeout set in config.
+	 * The information is resolved from headers (apache IntrospectionEndpoint call) and stored in principal's additionalInformations.
+	 * Check if the auth time + mfa timeout is not older than the current time
+	 * auth time = time of the first authentication
+	 * mfa timeout = amount of time defined in the config for how long the MFA should be valid (since SFA)
 	 *
 	 * @param sess session
+	 * @param throwError if this method should throw errors or just return boolean
 	 * @return true if principal authorized by MFA in allowed limit, false otherwise
 	 */
-	private static boolean isAuthorizedByMfa(PerunSession sess) {
+	private static boolean isAuthorizedByMfa(PerunSession sess, boolean throwError) {
```

Comment:
The `throwError` argument seems redundant. The caller know if exception needs to be thrown or not.

Reply:
I am looking at the code again and I think it is not probably redundant. In this method (`isAuthorizedByMfa()`) can be returned false in more cases:
```
if (!BeansUtils.getCoreConfig().isEnforceMfa()) {
	return false;
}
```
and
```
// check if the auth time + mfa timeout > the current time
if (mfaValidUntil.isAfter(Instant.now())) {
	// if user has MFA and it is still valid
	return sess.getPerunPrincipal().getAdditionalInformations().containsKey(ACR_MFA);
```
so without a duplicate check, I wouldn't be able to know from the caller when to throw an exception.